Metadata-Matcher/run_script.py

```diff
@@ -0,0 +1,168 @@
+import re
+from math import isnan
+from pathlib import Path
+import shutil
+import logging
+
+from utils import Sheet
+
+
+class Journal:
+    def __init__(self, config):
+        self.parent_dir = Path(config["parent_dir"])
+        self.metadata = self.parent_dir / config["metadata"]
+        self.folder = self.parent_dir / config["text_files_folder"]
+        self.destination_not_found = self.parent_dir / config["destination_not_found"]
+        self.destination_found = self.parent_dir / config["destination_found"]
+        self.use_abstract = config["use_abstract"]
+        self.files = []
+        self.config = config
+
+    def search(self):
+        self._scan_folder()
+        self._scan_metadata()
+        self._perform_match()
+
+    def _scan_folder(self):
+        """
+        Scan for the text files, retrieve the full paths
+        """
+        folder_path = Path(self.folder)
+        logging.info(
+            f"Scanning for '{self.config['text_files_glob']}' files inside '{folder_path}'")
+
+        text_files = list(folder_path.glob(self.config["text_files_glob"]))
+        self.files = text_files
+        logging.info(
+            f"Found {len(self.files)} '{self.config['text_files_glob']}' files")
+
+    def _scan_metadata(self):
+        """
+        Scan metadata sheet, retrieve all articles
+        """
+        logging.info(f"Scanning metadata - '{self.metadata}'")
+        self.sheet = Sheet(
+            str(self.metadata), self.config['columns'], use_abstract=self.use_abstract)
+        logging.info(
+            f"Found {len(self.sheet.articles)} rows inside the metadata sheet")
+
+    def _perform_match(self):
+        """
+        Match metadata and text files
+        """
+        logging.info("Starting metadata matching")
+
+        # Enumerate all text files
+        for i, text_file in enumerate(self.files):
+            logging.info(
+                f"Scanning File ({i + 1}/{len(self.files)}) {text_file}")
+            match_found = False
+
+            # Enumerate each row in metadata sheet
+            for article_meta in self.sheet.articles:
+                cutoff = 70 if self.use_abstract else 50
+                score = self._get_score_for_text(text_file, article_meta)
+                if score >= cutoff:
```

Comment:
what if two texts match the same metadata rows? will it just use the last one matched? if so, maybe you want to check which one has the higher matching score and then match the one with the higher score?